misc/python/materialize/mzcompose.py

```diff
@@ -619,18 +618,34 @@ def named(cls, name: str) -> Type["WorkflowStep"]:
             raise errors.UnknownItem("step", name, list(cls._steps))
 
     @classmethod
-    def register(cls, name: str) -> Callable[[Type[T]], Type[T]]:
+    def register(
+        cls, name: str
+    ) -> Callable[[Type["WorkflowStep"]], Type["WorkflowStep"]]:
         if name in cls._steps:
             raise ValueError(f"Double registration of step name: {name}")
 
-        def reg(to_register: Type[T]) -> Type[T]:
+        def reg(to_register: Type["WorkflowStep"]) -> Type["WorkflowStep"]:
```

Comment:
Was this causing problems? This type renaming is slightly worse in the abstract: it may make mypy believe that you have erased the type of the step that you have registered. The `T` should tell mypy that `register(obj)` returns exactly the same type of object that you passed to it.

Reply:
`mypy` was complaining that `T` is not guaranteed to have a `run()` method that was being used below,

Reply:
This doesn't sound right to me. mypy will now statically prevent you from calling `Step.register` on something that is not a `WorkflowStep`, which is correct. It also means we can remove the explicit `issubclass` check inside the function, since the type check is now enforced statically.